Revit_Core_Engine/Convert/Environment/ToRevit/Pipe.cs

```diff
@@ -0,0 +1,136 @@
+
+            Autodesk.Revit.DB.Plumbing.PipingSystemType pst = new FilteredElementCollector(document).OfClass(typeof(Autodesk.Revit.DB.Plumbing.PipingSystemType)).OfType<Autodesk.Revit.DB.Plumbing.PipingSystemType>().FirstOrDefault();
+
+            revitPipe = Autodesk.Revit.DB.Plumbing.Pipe.Create(document, pst.Id, pipeType.Id, level.Id, start, end);
+
+            // Copy parameters from BHoM object to Revit element
+            revitPipe.CopyParameters(pipe, settings);
+
+            SectionProfile sectionProfile = pipe.SectionProperty.SectionProfile;
+            if (sectionProfile == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Pipe creation requires a SectionProfile. \n No elements created.");
+                return null;
+            }
+
+            PipeSectionProperty pipeSectionProperty = pipe.SectionProperty;
+
+            // ToDo: Verify builtInParameter for correct value. This is not functioning as expected
+            double flowRate = pipe.FlowRate;
+            revitPipe.SetParameter(BuiltInParameter.RBS_DUCT_FLOW_PARAM, flowRate); //Not being set correctly.
+
+            // Round Pipe 
+            if (pipeType.Shape == ConnectorProfileType.Round)
+            {
+                TubeProfile elementProfile = sectionProfile.ElementProfile as TubeProfile;
+                if (elementProfile == null)
+                    return null;
+
+                // Set Element Diameter
+                double diameter = elementProfile.Diameter;
+                revitPipe.SetParameter(BuiltInParameter.RBS_PIPE_DIAMETER_PARAM, diameter);
+
+                // Pipe Lining is not supported in RevitAPI.
+
+                // Set InsulationProfile
+                TubeProfile insulationProfile = sectionProfile.InsulationProfile as TubeProfile;
+                if (insulationProfile != null)
+                {
+                    double insulationThickness = insulationProfile.Thickness;
+
+                    // Get first available pipeLiningType from document
+                    Autodesk.Revit.DB.Plumbing.PipeInsulationType pit = new FilteredElementCollector(document).OfClass(typeof(Autodesk.Revit.DB.Plumbing.PipeInsulationType)).FirstOrDefault() as Autodesk.Revit.DB.Plumbing.PipeInsulationType;
+                    if (pit == null)
+                    {
+                        BH.Engine.Reflection.Compute.RecordError("You must first create a pipe with insulation in your Revit model.");
+                        return null;
+                    }
+                    // Create pipe Insulation
+                    Autodesk.Revit.DB.Plumbing.PipeInsulation pIn = Autodesk.Revit.DB.Plumbing.PipeInsulation.Create(document, revitPipe.Id, pit.Id, insulationThickness);
+                }        
+                // Parameters not found in RevitAPI for Pipes, needs further investigation. Not critical for current workflows. 
+                // double profileThickness = elementProfile.Thickness;
```

Comment:
i think this one comes from the inner and outer diameter (there are such params in the API)